### PR TITLE
refactor(release): update release github action

### PR DESCRIPTION
**Pull request by solareon**
Modifies lint action to only run on changes to lua files. Replaces release action and change log generation with newer and supported actions.

### CODE REVIEW

**Review by ChatDisabled — APPROVED** [review]
Cool 🤘

File: repo-template/.github/workflows/lint.yml [hunk 1]
```diff
@@ -1,5 +1,13 @@
...
-on: [push, pull_request_target]
+on:
+    push:
+      paths:
+        - '*.lua'
+        - '.github/workflows/lint.yml'
+    pull_request_target:
+      paths:
+        - '*.lua'
+        - '.github/workflows/lint.yml'
```
Comment (D4isDAVID) [inline repo-template/.github/workflows/lint.yml]:
```suggestion
  push:
    paths:
      - '*.lua'
      - '.github/workflows/lint.yml'
  pull_request_target:
    paths:
      - '*.lua'
      - '.github/workflows/lint.yml'
```
Not sure how much this indentation affects YAML but we should keep it the same.

File: repo-template/.github/workflows/release-action.yml [hunk 2]
```diff
@@ -34,10 +34,24 @@ jobs:
...
+            - name: Update CHANGELOG
+              id: changelog
+              uses: requarks/changelog-action@v1
+              with:
+                token: ${{ steps.generate_token.outputs.token }}
+                tag: ${{ github.ref_name }}
+                includeInvalidCommits: true
+                writeToFile: false
+
+
```
Comment (D4isDAVID) [inline repo-template/.github/workflows/release-action.yml]:
```suggestion
```
Nitpick

**Review by D4isDAVID — CHANGES_REQUESTED** [review]

**Comment by mafewtm** [comment]
I vote keeping the changelogs how they are now instead but thats just me

**Comment by D4isDAVID** [comment]
Isn't this the same just auto-generated?

**Comment by solareon** [comment]
The old ones were auto generated as well it was just baked into the `marvinpinto/action-automatic-releases` action vice being a separate step

**Review by mafewtm — APPROVED** [review]

**Review by D4isDAVID — APPROVED** [review]

**Pull request merged**